exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1966 @@
+	}
+	if o.IcebergQty != 0 {
+		params.Set("icebergQty", strconv.FormatFloat(o.IcebergQty, 'f', -1, 64))
+	}
+	if o.NewOrderRespType != "" {
+		params.Set("newOrderRespType", o.NewOrderRespType)
+	}
+	return bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodPost, api, params,
+		spotOrderRate, resp)
+}
+
+// GetOrder to check a trade order's status.
+func (bi *Binanceus) GetOrder(ctx context.Context, arg *OrderRequestParams) (*Order, error) {
+	var resp Order
+	params := url.Values{}
+	if arg.Symbol == "" {
+		return nil, errIncompleteArguments
+	}
+	params.Set("symbol", strings.ToUpper(arg.Symbol))
+	if arg.OrderID > 0 {
+		params.Set("orderId", strconv.Itoa(int(arg.OrderID)))
+	}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	if arg.OrigClientOrderID != "" {
+		params.Set("origClientOrderId", arg.OrigClientOrderID)
+	}
+	if arg.recvWindow > 200 && arg.recvWindow <= 6000 {
+		params.Set("recvWindow", strconv.Itoa(int(arg.recvWindow)))
+	}
+	return &resp, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodGet, orderRequest,
+		params, spotDefaultRate,
+		&resp)
+}
+
+// GetAllOpenOrders to get all open trade orders on a token symbol. Do not access this without a token symbol as this would return all pair data.
+func (bi *Binanceus) GetAllOpenOrders(ctx context.Context, symbol string) ([]*Order, error) {
+	var response []*Order
+	params := url.Values{}
+
+	timestamp := time.Now().UnixMilli()
+	if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	params.Set("recvWindow", recvWindowSize5000String)
+	var rateLimit request.EndpointLimit
+	if symbol != "" {
+		rateLimit = spotOpenOrdersSpecificRate
+	} else {
+		rateLimit = spotOpenOrdersAllRate
+	}
+	return response, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary, http.MethodGet,
+		openOrders, params,
+		rateLimit, &response)
+}
+
+// CancelExistingOrder to cancel an active trade order.
+func (bi *Binanceus) CancelExistingOrder(ctx context.Context, arg *CancelOrderRequestParams) (*Order, error) {
+	params := url.Values{}
+	var response Order
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
+	symbolValue, err := bi.FormatSymbol(arg.Symbol, asset.Spot)
+	if err != nil || symbolValue == "" {
+		return nil, errMissingCurrencySymbol
+	}
+	params.Set("symbol", symbolValue)
+	if arg.OrderID == "" && arg.ClientSuppliedOrderID == "" {
+		return nil, errEitherOrderIDOrClientOrderIDIsRequired
+	}
+	if arg.ClientSuppliedOrderID != "" {
+		params.Set("origClientOrderId", arg.ClientSuppliedOrderID)
+	} else {
+		params.Set("orderId", arg.OrderID)
+	}
+	params.Set("recvWindow", recvWindowSize5000String)
+	return &response, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodDelete, orderRequest,
+		params, spotDefaultRate, &response)
```

Comment:
```suggestion
		params, spotOrderRate, &response)
```

Reply:
The weight of this endpoint is 2 and SpotOrderRate has a rate limit of 1 so that's why I  created 'spotOrderQueryRate' and gave it a weight of 2

Reply:
But its still `spotDefaultRate`? And cancel orders is a weight of 1: https://docs.binance.us/#get-all-open-orders. Was your reply for a different comment?

The purpose is to move order related rate limits to the separate order rate limits of `SpotOrdersRate` instead of `SpotRate`